src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1805,6 +1805,22 @@ namespace winrt::TerminalApp::implementation
             {
                 ShellExecute(nullptr, L"open", eventArgs.Uri().c_str(), nullptr, nullptr, SW_SHOWNORMAL);
             }
+            if (parsed.SchemeName() == L"file")
+            {
+                const auto host = parsed.Host();
+                // If no hostname was provided or if the hostname was "localhost", Host() will return null
+                // and we allow it
+                if (host == L"")
+                {
```

Comment:
these can probably be folded into a single condition, or maybe abstracted out to a helper function ("IsUrlSupported")

That will make it easier to add additional fail-out conditions in #7523 -- otherwise, you'll need to restructure these nested ifs/elses to pop up the dialog anyway

Reply:
Fixed! Chose the helper function route